lib/galaxy/tool_util/xsd/galaxy.xsd

```diff
@@ -3725,21 +3732,28 @@ use in filenames may not contain ``..``.
       <xs:extension base="xs:string">
         <xs:attribute name="type" type="ValidatorType" use="required">
           <xs:annotation>
-            <xs:documentation xml:lang="en"><![CDATA[The list of supported
+            <xs:documentation xml:lang="en"><![CDATA[
+Valid values are: ``expression``, ``regex``, ``in_range``, ``length``,
+``metadata``, ``unspecified_build``, ``no_options``, ``empty_field``,
+``dataset_metadata_in_data_table``, ``value_in_data_table``,
+``dataset_ok_validator``, ``dataset_metadata_in_range``.
+Deprecated validators: ``dataset_metadata_in_file``, ``value_not_in_data_table``, ``dataset_metadata_not_in_data_table``.
+The list of supported
 validators is in the ``validator_types`` dictionary in
 [/lib/galaxy/tools/parameters/validation.py](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/parameters/validation.py).
-Valid values include: ``expression``, ``regex``, ``in_range``, ``length``,
-``metadata``, ``unspecified_build``, ``no_options``, ``empty_field``,
-``dataset_metadata_in_file``,
-``dataset_metadata_in_data_table``, ``dataset_metadata_not_in_data_table``,
-``value_in_data_table``, ``value_not_in_data_table``,
-``dataset_ok_validator``, ``dataset_metadata_in_range``]]></xs:documentation>
+]]></xs:documentation>
           </xs:annotation>
         </xs:attribute>
         <xs:attribute name="message" type="xs:string">
           <xs:annotation>
             <xs:documentation xml:lang="en">
-The error message displayed on the tool form if validation fails.</xs:documentation>
+The error message displayed on the tool form if validation fails. A placeholder string ``%s`` will be repaced by the ``value``</xs:documentation>
```

Comment:
> Do we have an example for that ?

Do you mean a functional tool test that we could link?

> I'll have to admit I only know what %s and value are because I saw the docstring tests you added. 

I guess the formulation can be improved.

> Does this work for all parameter types, e.g collections and datasets ?

Probably this would lead to undesired behavior. Maybe we can check the type of `value` is and print the name of value if its a dataset of a collection?

Reply:
> Do you mean a functional tool test that we could link?

Yeah, that would work.

> Maybe we can check the type of `value` is and print the name of value if its a dataset of a collection?

That sounds good, but I'm a bit nervous about accidentally leaking out internals via the string substitution and binding ourselves so closely to python syntax within tool parameters. To some degree it feels a bit redundant to even substitute in the value ... the frontend knows which value raised the validation error. But I guess this might be useful when running workflows too ? ... @jmchilton do you have any thoughts on this ?